### DISCUSSION TITLE
filter flowrate sensor

### DISCUSSION THREAD

**Discussion by pet-valent** [davidm-glitch/home-assistant-eheim-digital#19] [Ideas]
Hi it seems that after update to 1.01.8 (web 1.02.0) in overview for 5e 700 filter info about actual flowrate in l/h is available next to the polution percentage, would it be possible to add this as a sensor?

**Comment by jkhalil** [comment]
Hi @pet-valent, can you post a screen shot? I actually don't have a device and have been developing this off of @davidm-glitch's device. I don't think he has upgraded the software yet so I am unable to see what data I need to grab and create the sensor.

Are you technical enough to do some logging for us and post the data here? I outlined how to do it here:  #5. 

If you can, you would just need to  look for anything that looks like:

 `{"title":"GET_FILTER_DATA","to":"E8:DB:84:88:5F:81","from":"USER"}`

 and the response after/above that call should look something like this:
_They will be in rows_ like the picture below
`{
    "title": "FILTER_DATA",
    "from": "E8:DB:84:88:5F:81",
    "minFreq": 3500,
    "maxFreq": 7500,
    "maxFreqRglOff": 8000,
    "freq": 6050,
    "freqSoll": 6050,
    "dfs": 0,
    "dfsFaktor": 0,
    "sollStep": 8,
    "rotSpeed": 36,
    "pumpMode": 16,
    "filterActive": 1,
    "runTime": 360306,
    "actualTime": 180256,
    "serviceHour": 4320,
    "pm_dfs_soll_high": 0,
    "pm_dfs_soll_low": 0,
    "pm_time_high": 5,
    "pm_time_low": 5,
    "nm_dfs_soll_day": 7,
    "nm_dfs_soll_night": 9,
    "end_time_night_mode": 720,
    "start_time_night_mode": 1320,
    "version": 78,
    "isEheim": 0,
    "turnOffTime": 0,
    "turnTimeFeeding": 0,
    "to": "USER"
}`

![image](https://github.com/davidm-glitch/home-assistant-eheim-digital/assets/3517333/7aeed0b1-8165-4668-b524-ff49e8e3d269)

**Comment by pet-valent** [comment]
![image](https://github.com/davidm-glitch/home-assistant-eheim-digital/assets/75450842/ccf49113-fc8e-4834-9a6c-b01c2f14f094)

**Reply by jkhalil** [reply]
Interesting... I wonder if it's just a calculation of the rotor speeds, what was there before.

**Comment by pet-valent** [comment]
I will try to retrieve the data one when I am on home PC

**Comment by pet-valent** [comment]
So looking and comparing of data seems to be some kind of calculation.
Flowrate that was shown was 508l/h with filter set to bio mode and flowrate set from night timming 785l/h

data:
        "minFreq": 3500,
        "maxFreq": 7500,
        "maxFreqRglOff": 8000,
        "freq": 7500,
        "freqSoll": 7500,
        "dfs": 845,
        "dfsFaktor": 1129,
        "sollStep": 7,
        "rotSpeed": 45,
        "pumpMode": 4,
        "filterActive": 1,
        "runTime": 1087652,
        "actualTime": 54222,
        "serviceHour": 4760,
        "pm_dfs_soll_high": 11,
        "pm_dfs_soll_low": 1,
        "pm_time_high": 10,
        "pm_time_low": 10,
        "nm_dfs_soll_day": 10,
        "nm_dfs_soll_night": 7,
        "end_time_night_mode": 360,
        "start_time_night_mode": 1140,
        "version": 78,
        "isEheim": 0,
        "turnOffTime": 0,
        "turnTimeFeeding": 0
